vnext/ReactUWP/Modules/Animated/NativeAnimatedNodeManager.h

```diff
@@ -0,0 +1,68 @@
+#pragma once
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Portions derived from React Native:
+// Copyright (c) 2015-present, Facebook, Inc.
+// Licensed under the MIT License.
+
+#include <folly/dynamic.h>
+#include "AnimatedNode.h"
+#include "PropsAnimatedNode.h"
+#include "ValueAnimatedNode.h"
+#include "AnimationDriver.h"
+#include "TransformAnimatedNode.h"
+#include "EventAnimationDriver.h"
+#include <cxxreact/CxxModule.h>
+#include <IReactInstance.h>
+
+namespace react { namespace uwp {
+  /// <summary>
+  /// This is the main class that coordinates how native animated JS
+  /// implementation drives UI changes.
+  /// 
+  /// It implements a management interface for animated nodes graph and
+  /// establishes a number of composistion animations and property sets to
+  /// drive the animating of the nodes to the Xaml elements off the UI Thread
+  /// 
+  /// </summary>
+
+  typedef std::function<void(std::vector<folly::dynamic>)> Callback;
+
+  class StyleAnimatedNode;
+  class PropsAnimatedNode;
+  class ValueAnimatedNode;
+  class TransformAnimatedNode;
+  class AnimationDriver;
+  class EventAnimationDriver;
+  class NativeAnimatedNodeManager
+  {
+  public:
+    void CreateAnimatedNode(int64_t tag, const folly::dynamic& config, const std::weak_ptr<IReactInstance>& instance, const std::shared_ptr<NativeAnimatedNodeManager>& manager);
+    void ConnectAnimatedNodeToView(int64_t propsNodeTag, int64_t viewTag);
+    void DisconnectAnimatedNodeToView(int64_t propsNodeTag, int64_t viewTag);
+    void ConnectAnimatedNode(int64_t parentNodeTag, int64_t childNodeTag);
+    void DisconnectAnimatedNode(int64_t parentNodeTag, int64_t childNodeTag);
+    void StopAnimation(int64_t animationId);
+    void StartAnimatingNode(int64_t animationId, int64_t animatedNodeTag, const folly::dynamic& animationConfig, const Callback& endCallback);
+    void DropAnimatedNode(int64_t tag);
+    void SetAnimatedNodeValue(int64_t tag, double value);
+    void SetAnimatedNodeOffset(int64_t tag, double offset);
+    void FlattenAnimatedNodeOffset(int64_t tag);
+    void ExtractAnimatedNodeOffset(int64_t tag);
+    void AddAnimatedEventToView(int64_t viewTag, const std::string& eventName, const folly::dynamic& eventMapping);
+    void RemoveAnimatedEventFromView(int64_t viewTag, const std::string& eventName, int64_t animatedValueTag);
+    void ProcessDelayedPropsNodes();
+    void AddDelayedPropsNode(int64_t propsNodeTag, const std::shared_ptr<IReactInstance>& instance);
+
+    std::unordered_map<int64_t, std::shared_ptr<AnimationDriver>> m_animationNodes{};
```

Comment:
are all the shared_ptr<AnimationNode*> types actually needing ref counting for multiple owners or thread safety?  It looks like maybe these can just be unique_ptr or an inline instance and GetAnimatedNode could return a reference ?